src/boxerror_replacement.rs

```diff
@@ -1,13 +1,25 @@
 //! Experimental replacement for `Box<dyn Error>` that implements `Error`.
 //!
-//! This module introduces the `DynError` type, which owns an inner 
-//! `Box<dyn Error>`. Most importantly, the `DynError` type implements the `Error`
-//! trait, unlike a plain `Box<dyn Error>`. Another important distinction between
-//! `DynError` and `Box<dyn Error>` is that `DynError` _doesn't_ implement 
-//! `From<E: Error>`. 
+//! This module introduces the `DynError` type, which owns an inner `Box<dyn Error>`. Most
+//! importantly, `DynError` _does_ implement the `Error` trait, unlike `Box<dyn Error>`. As a
+//! result, `DynError` is more conveniently compatible with the rest of the error handling
+//! ecosystem, and behaves as any other error type does. 
+//!
+//! The short answer as to why `Box<dyn Error>` doesn't implement `Error` is because there 
+//! exists a blanket implementation of the `Error` trait for `Box<T>`, more specifically: 
+//! `impl<T: Error + Sized> Error for Box<T>`. Crucially, any type `T` must be sized. However, 
+//! when `T` is a `dyn Error` trait object, it is _not_ sized. For more context on why 
+//! `Box<dyn Error>` doesn't implement the `Error` trait, see 
+//! https://stackoverflow.com/questions/65151237/why-doesnt-boxdyn-error-implement-error. 
+//!
+//! Another important distinction between `DynError` and `Box<dyn Error>` is that `DynError`
+//! _doesn't_ implement `From<E: Error>`, as this would violate the Overlap Rule. As a result of
+//! this, `DynError`s must be mapped to a `DynResult` first, which can then be manually converted
+//! into a `Result`. This is a bit of an ergonomic hit; it also introduces a bit of runtime
+//! overhead as converting from a `DynError` to a `DynResult` requires downcasting at runtime.
```

Comment:
I would reframe this as "how we avoided this issue while still allowing ? to work with any Error type" rather than just as another difference.